sqlmesh/integrations/github/cicd/controller.py

```diff
@@ -0,0 +1,717 @@
+        not update the comment. If no `find_regex` is provided then it will just append the value to the comment.
+        """
+        comment_header = "**SQLMesh Bot Info**"
+        comment = seq_get(
+            [comment for comment in self._issue.get_comments() if comment_header in comment.body],
+            0,
+        )
+        if not comment:
+            comment = self._issue.create_comment(comment_header)
+        existing_value = seq_get(re.findall(find_regex, comment.body), 0) if find_regex else None
+        if existing_value:
+            if replace_if_exists:
+                comment.edit(re.sub(t.cast(str, find_regex), value, comment.body))
+        else:
+            comment.edit(f"{comment.body}\n{value}")
+
+    def update_pr_environment(self) -> None:
+        """
+        Creates a PR environment from the logic present in the PR. If the PR contains changes that are
+        uncategorized, then an error will be raised.
+        """
+        self._context.apply(self.pr_plan)
+
+    def deploy_to_prod(self) -> None:
+        """
+        Attempts to deploy a plan to prod. If the plan is not up-to-date or has gaps then it will raise.
+        """
+        # If the PR is already merged then we will not deploy to prod if this event was triggered prior to the merge.
+        # The deploy can still happen if the workflow is configured to listen for `closed` events.
+        if self._pull_request.merged and not self._event.is_pull_request_closed:
+            raise CICDBotError(
+                "PR is already merged and this event was triggered prior to the merge. So not running."
+            )
+        plan_summary = f"""<details>
+  <summary>Prod Plan Being Applied</summary>
+
+{self._get_plan_summary(self.prod_plan)}
+</details>
+
+"""
+        self.update_sqlmesh_comment_info(
+            value=plan_summary,
+            find_regex=None,
+        )
+        self._context.apply(self.prod_plan)
+
+    def delete_pr_environment(self) -> None:
+        """
+        Deletes all the schemas for a given environment by checking all the schemas used by models.
+        """
+        schemas = {parse_model_name(model.name)[1] for model in self._context.models.values()}
+        for schema in schemas:
+            assert schema
+            self._context.engine_adapter.drop_schema(
+                schema_name="__".join([schema, self.pr_environment_name]),
+                ignore_if_not_exists=True,
+                cascade=True,
+            )
+        return
+
+    def get_pr_affected_models(self) -> t.List[AffectedEnvironmentModel]:
+        plan = self._context.plan(
+            c.PROD,
+            auto_apply=False,
+            no_gaps=False,
+            no_prompts=True,
+            no_auto_categorization=True,
+            skip_tests=True,
+        )
+        affected_env_models = []
+        for snapshot in plan.directly_modified:
```

Comment:
this feels like it should be a method on the plan, not something specific to github

Reply:
It has specific logic for filtering out indirect non-breaking from the environment summary. We don't produce these "environment" summaries anywhere else.

Reply:
it feels generally reusable, and you'd use it in a future integration, let's move it to plan

Reply:
See https://github.com/TobikoData/sqlmesh/pull/810/commits/e868b0a51b6b2a35edf835267d204fca427e2914